### PR TITLE
[TS] LPS-145103

### PR DESCRIPTION
**Pull request by istvansajtos**
Hi Lima Team,

Please review my changes and let me know if you have any questions or concerns.

Thank you,
Istvan

### CODE REVIEW

**Comment by istvansajtos** [comment]
CI test results are available here: https://github.com/liferay-echo/liferay-portal/pull/6901

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 1]
```diff
@@ -176,6 +180,23 @@ public Redirect getRedirect(
 
 			Layout layout = layoutFriendlyURLSeparatorComposite.getLayout();
 
+			User user = portal.getUser(httpServletRequest);
+
+			if (user == null) {
+				user = userLocalService.getDefaultUser(
+					portal.getCompanyId(httpServletRequest));
+			}
```
Comment (adolfopa) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java:188]: Can you move this to a private _getUser() method?

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java [hunk 2]
```diff
@@ -176,6 +180,23 @@ public Redirect getRedirect(
...
+
+			PermissionChecker permissionChecker =
+				PermissionCheckerFactoryUtil.create(user);
+
+			boolean hasViewPermission = LayoutPermissionUtil.contains(
+				permissionChecker, layout, ActionKeys.VIEW);
+
+			if (!hasViewPermission) {
+				throw new NoSuchLayoutException();
+			}
```
Comment (adolfopa) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/internal/servlet/FriendlyURLServlet.java:198]: Can you inline the hasViewPermission variable?

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by istvansajtos** [comment]
Thanks @adolfopa! I've applied the requested changes. Please let me know if any other improvement is required.

**Comment by istvansajtos** [comment]
ci:test:sf

**Comment by istvansajtos** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**